Update documentation for synced sandboxes. (#134)
codesandbox/docs: packages/projects-docs/pages/learn/sandboxes/overview.mdx

configuration like custom webpack configurations or ejecting. However, they are
designed to mirror the default behavior of the framework. 

> If your project requires advanced configuration, try using a [Cloud Sandbox](/learn/sandboxes/overview?tab=cloud) instead.

Reviewer: Advanced features like? 
```suggestion
If your project requires advanced configuration, try using a [Cloud Sandbox](/learn/sandboxes/overview?tab=cloud) instead.
```

       - Branching
       - Git tooling 
       - Full integration with GitHub
       
        ## What is a Synced Sandbox?

>         With CodeSandbox, you can import any GitHub repository, subfolder or branch as a synced cloud sandbox. 
>         
>         An imported sandbox will automatically stay in sync with the GitHub repository; this means that if you make a commit to GitHub, it will reflect immediately in the sandbox.

Reviewer: ```suggestion
A Synced sandbox is a Sandbox that stays up to date with the latest changes you have pushed to a GitHub project using your `.git` file.

Their main advantage is that they allow you to create sandboxes from specific sub-folders within your GitHub projects without importing the whole repo. 

They automatically stay in sync with the GitHub repository folder; this means that if you make a commit to GitHub, it will reflect immediately in the sandbox.
```

>         For that reason, we've made these GitHub synced sandboxes immutable. This means that you cannot make direct changes to the sandbox itself. 
>         
>         These sandboxes are then treated as [templates](/learn/sandboxes/templates), so you can fork and start a new project from them. When you create a fork of a GitHub sandbox, we will still keep a reference to the original GitHub repository. 

Reviewer: ```suggestion
For that reason, we've made these GitHub synced sandboxes immutable. This means that you cannot make direct changes to the sandbox itself. 

These sandboxes are then treated as [templates](/learn/sandboxes/templates), so you can fork and start a new project from them. 
```
